cluster/src/main/java/io/atomix/cluster/messaging/impl/NettyUnicastService.java

```diff
@@ -78,11 +80,17 @@ public NettyUnicastService(Address address, MessagingConfig config) {
 
   @Override
   public void unicast(Address address, String subject, byte[] payload) {
+    final InetAddress resolvedAddress = address.address();
+    if (resolvedAddress == null) {
+      LOGGER.debug("Cannot send unicast message to destination address {} as it cannot be resolved", address);
+      return;
+    }
+
     Message message = new Message(this.address, subject, payload);
     byte[] bytes = SERIALIZER.encode(message);
     ByteBuf buf = channel.alloc().buffer(4 + bytes.length);
     buf.writeInt(bytes.length).writeBytes(bytes);
-    channel.writeAndFlush(new DatagramPacket(buf, new InetSocketAddress(address.address(), address.port())));
```

Comment:
looks like this may fix a memory leak if writeAndFlush does not consume the bytebuf, nice!